src/main/webapp/app/shared/participant-scores/participant-scores-distribution/participant-scores-distribution.component.ts

```diff
@@ -0,0 +1,318 @@
+                label += i === this.ngxData.length - 1 ? ']' : ')';
+
+                this.ngxData[i].name = label;
+            }
+        }
+
+        this.ngxData = [...this.ngxData];
+    }
+
+    /**
+     * Find the grade step index for the corresponding grade step to the given percentage
+     * @param percentage the percentage which will be mapped to a grade step.
+     */
+    findGradeStepIndex(percentage: number): number {
+        let index = 0;
+        if (this.gradingScaleExists) {
+            this.gradingScale!.gradeSteps.forEach((gradeStep, i) => {
+                if (this.gradingSystemService.matchGradePercentage(gradeStep, percentage)) {
+                    index = i;
+                }
+            });
+        } else {
+            index = Math.floor(percentage / this.binWidth);
+            if (index >= 100 / this.binWidth) {
+                // This happens, for 100%, if the exam total points were not set correctly or bonus points were given
+                index = 100 / this.binWidth - 1;
+            }
+        }
+        return index;
+    }
+
+    /**
+     * Auxiliary method that increases the represented value of the bar covering the given score by 1
+     * @param percentage the score that should be included in the distribution
+     * @private
+     */
+    private addToHistogram(percentage: number): void {
+        // Update histogram data structure
+        const histogramIndex = this.findGradeStepIndex(percentage);
+        this.ngxData[histogramIndex].value++;
+    }
+
+    /**
+     * Sets up the distribution chart for given scores
+     * @private
+     */
+    private createChart(): void {
+        this.generateDefaultNgxChartsSetting();
+        this.setupChartColoring();
+        this.setupAxisLabels();
+        this.scores.forEach((score) => this.addToHistogram(score));
+    }
+
+    /**
+     * Determines and returns the maximum value displayed on the Y axis
+     * @private
+     */
+    private calculateTickMax(): number {
+        const histogramData = this.ngxData.map((dataPack) => dataPack.value);
+        const max = Math.max(...histogramData);
+        return Math.ceil((max + 1) / 10) * 10 + 20;
+    }
+
+    /**
+     * Determines and returns the translation path for the tooltip that is displayed when hovering over the help icon next to the legend
+     * @private
+     */
+    private determineHelpIconTooltip(): string {
+        if (this.gradingScaleExists) {
+            if (this.isBonus) {
+                return 'artemisApp.examScores.gradingScaleExplanationBonus';
+            } else {
+                return this.getHelpIconTooltipNotBonus();
+            }
+        } else {
+            return this.getHelpIconNoGradingScale();
+        }
+    }
+
+    /**
+     * Auxiliary method that returns the bar color of the grade step in the chart
+     * @param gradeStep the grade step that should be colored
+     * @returns string representation of the color
```

Comment:
```suggestion
     * @returns the color that the given grade step should receive in the chart
```
The type annotation says that a GraphColor is returned.
(Of course the enum value is actually a string, but that is an implementation detail which is not relevant here.)